core/src/wsv.rs

```diff
@@ -298,16 +294,22 @@ impl WorldStateView {
             );
         }
 
-        self.blocks.push(block);
+        self.block_hashes.borrow_mut().push(block.hash());
 
         Ok(())
     }
```

Comment:
I don't see that block is ever sent to `kura`, is this possible?

Reply:
The block is put inside kura after this is called inside sumeragi. It's done in a synchronous manner so that there can't be a race condition. See `commit_block` inside main_loop.rs